frontend/src/components/IstioWizards/K8sRequestRouting.tsx

```diff
@@ -163,10 +177,10 @@ class K8sRequestRouting extends React.Component<Props, State> {
 
   onHeaderNameChange = (headerName: string) => {
     let validationMsg = '';
-    if (this.state.matchValue !== '' && headerName === '') {
+    if ((this.state.matchValue !== '' || this.state.headerOp === REMOVE) && headerName === '') {
```

Comment:
```suggestion
    if (!headerName && (!!this.state.matchValue || this.state.headerOp === REMOVE)) {
```

No change required but I think it's often better to use truthy/falsy to cover corner cases like `undefined` and such.